rasa/core/processor.py

```diff
@@ -264,23 +266,32 @@ def _has_message_after_reminder(
         ):
             logger.debug(
                 "Canceled reminder because it is outdated. "
-                "(event: {} id: {})".format(
-                    reminder_event.action_name, reminder_event.name
-                )
+                "(event: {} id: {})".format(reminder_event.intent, reminder_event.name)
             )
         else:
-            # necessary for proper featurization, otherwise the previous
-            # unrelated message would influence featurization
-            tracker.update(UserUttered.empty())
-            action = self._get_action(reminder_event.action_name)
-            should_continue = await self._run_action(
-                action, tracker, output_channel, nlg
+            intent_name = reminder_event.intent
+            await self.inject_external_user_uttered(
+                intent_name, None, tracker, output_channel
             )
-            if should_continue:
-                user_msg = UserMessage(None, output_channel, sender_id)
-                await self._predict_and_execute_next_action(user_msg, tracker)
-            # save tracker state to continue conversation from this state
-            self._save_tracker(tracker)
+
+    async def inject_external_user_uttered(
+        self,
+        intent_name: Text,
+        entities,
+        tracker: DialogueStateTracker,
+        output_channel: OutputChannel,
+    ):
+        tracker.update(
+            UserUttered(
+                text=f"{EXTERNAL_MESSAGE_PREFIX}{intent_name}",
```

Comment:
is there a reason for using this prefix setup instead of an attribute on this class?

Reply:
The idea was that you have this text as placeholder for the external event when you're showing conversations.

Reply:
So we keep this? If the user doesn't want to see the prefix, (s)he can just set `EXTERNAL_MESSAGE_PREFIX` to an empty string.

Reply:
We could also extend the class or add a static method `UserUttered.create_external(intent_name, entity_list)`. This way we keep the responsibility in the event instead of having it here in the `processor`